File: Algorithms/Search Algorithm/BinarySearch/binarySearch.go
package binarysearch

func search(arr []int, needle int)bool{
    lo := 0
    hi := len(arr) - 1

    for lo < hi{
        mid := (lo + hi) / 2
        if arr[mid] == needle{
            return true
        }
        if arr[mid] < needle{
            lo = mid + 1
        }else{
            hi = mid
        }
    }

    return false
}
func main(){}

File: Algorithms/Search Algorithm/BinarySearch/binarySearch.ts
const search = (arr:number[],needle:number):boolean=>{
    let lo = 0
    let hi = arr.length-1

    do {
       const middle = Math.floor((lo+hi)/2)
       
        if(arr[middle]===needle) return true
        if(arr[middle]<needle) lo = middle+1
        else hi = middle
    
    } while (lo<hi);

    return false
}

